components/NavigationDrawer/tests/snapshot/MDCBottomDrawerControllerSnapshotTests.m

```diff
@@ -0,0 +1,127 @@
+// Copyright 2019-present the Material Components for iOS authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#import "MaterialSnapshot.h"
+
+#import "../../src/private/MDCBottomDrawerContainerViewController.h"
+#import "MaterialNavigationDrawer.h"
+#import "supplemental/MDCBottomDrawerSnapshotTestMutableTraitCollection.h"
+
+/** Fake MDCBottomDrawerContainerViewController for snapshot testing. */
+@interface FakeContainerViewController : MDCBottomDrawerContainerViewController
+
+/** Used to set the value of @c traitCollection. */
+@property(nonatomic, strong) UITraitCollection *traitCollectionOverride;
+
+@end
+
+@implementation FakeContainerViewController
+
+- (UITraitCollection *)traitCollection {
+  return self.traitCollectionOverride ?: [super traitCollection];
+}
+
+@end
+
+/** Fake view controller for the MDCBottomDrawer header for snapshot testing. */
+@interface FakeHeader : UIViewController <MDCBottomDrawerHeader>
```

Comment:
Tip: I'm not sure, but I suspect you should prefix this class name with something about the component. Otherwise another component could create a `FakeHeader` class and the linker will observe duplicate symbols.

Reply:
Done, thanks